Fix compiling and running issue on aarch64

Jianlin-lv opened the pull request:
Add header files path of libc6-dev:
    /usr/include/aarch64-linux-gnu/
to support compiling on aarch64 platform

Skip checking header file code on arm64 platform.

Fixes: #9898

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9900)
<!-- Reviewable:end -->

Jianlin-lv committed:
loader:Fix compiling issue of bpf_lxc.c on aarch64

Add header files path of libc6-dev:
    /usr/include/aarch64-linux-gnu/
to support compiling on aarch64 platform

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

Jianlin-lv committed:
linux: Fix running issue on aarch64

Skip checking header file code on arm64 platform

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

aanm commented on pkg/datapath/loader/compile.go:
```diff
@@ -192,6 +194,17 @@ func compileAndLink(ctx context.Context, prog *progInfo, dir *directoryInfo, deb
+	var unameBuf unix.Utsname
+	err := unix.Uname(&unameBuf)
+	if err != nil {
+		log.WithError(err).Fatal("Get machine hardware name error!")
```
Please do not fatal here, return the error to the caller.

Jianlin-lv replied:
Corrected by https://github.com/cilium/cilium/pull/9900/commits/73ea3c61dac32e5f8c9694f157070a40223a3d4e

aanm commented on pkg/datapath/linux/requirements.go:
```diff
@@ -184,17 +185,24 @@ func CheckMinRequirements() {
-			log.Fatal("linking environment: NOT OK, please make sure you have 'libc6-dev-i386' in your ubuntu system")
+		hardwarename, err := loader.GetHardwareName()
+		if err != nil {
+			log.WithError(err).Fatal("Get machine hardware name error!")
```
I would be careful with the fatal here. Maybe an error would be good enough? In the end, this whole section is for people that are not running cilium in a container because our container image already has all the dependencies installed.

Jianlin-lv replied:
Corrected by https://github.com/cilium/cilium/pull/9900/commits/73ea3c61dac32e5f8c9694f157070a40223a3d4e

Jianlin-lv committed:
linux: Fix running issue on aarch64

Enable validation for linking environment by
checking state of the header file and update log info.

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

tklauser reviewed (changes requested):
With #10204 being merged, I think this PR is no longer needed since `/usr/include/$(uname -m)-linux-gnu/` is now always part of the include search path. @Jianlin-lv Can you please check?
